install/installer/pkg/components/ws-proxy/objects.go

```diff
@@ -14,23 +17,33 @@ var Objects = common.CompositeRenderFunc(
 	networkpolicy,
 	rolebinding,
 	role,
+	func(cfg *common.RenderContext) ([]runtime.Object, error) {
+		ports := map[string]common.ServicePort{
+			HTTPProxyPortName: {
+				ContainerPort: HTTPProxyPort,
+				ServicePort:   HTTPProxyPort,
+			},
+			HTTPSProxyPortName: {
+				ContainerPort: HTTPSProxyPort,
+				ServicePort:   HTTPSProxyPort,
+			},
+			MetricsPortName: {
+				ContainerPort: MetricsPort,
+				ServicePort:   MetricsPort,
+			},
+			SSHPortName: {
+				ContainerPort: SSHTargetPort,
+				ServicePort:   SSHServicePort,
+			},
+		}
+		return common.GenerateService(Component, ports, func(service *corev1.Service) {
+			// In the case of Workspace only setup, `ws-proxy` service is the entrypoint
+			// Hence we use LoadBalancer type for the service
+			if cfg.Config.Kind == config.InstallationWorkspace {
+				service.Spec.Type = corev1.ServiceTypeLoadBalancer
+				service.Annotations["cloud.google.com/neg"] = `{"exposed_ports": {"80":{},"443": {}}}`
```

Comment:
Do we need to hardcode google specific annotations here?

Reply:
These annotations have been hardcoded for some time on the basis that a) they're ignored if not used and b) we don't really have an effective (and non-verbose) way of adding custom annotations to services. Elsewhere, there's also an annotation for `external-dns` with the same thinking